core/web/health_controller.go

```diff
@@ -93,7 +99,200 @@ func (hc *HealthController) Health(c *gin.Context) {
 			Output: output,
 		})
 	}
+	slices.SortFunc(checks, func(a, b presenters.Check) int {
+		return cmp.Compare(a.Name, b.Name)
+	})
+
+	switch c.NegotiateFormat(gin.MIMEJSON, gin.MIMEHTML, gin.MIMEPlain) {
+	case gin.MIMEJSON:
+		break // default
+
+	case gin.MIMEHTML:
+		if err := newCheckTree(checks).WriteHTMLTo(c.Writer); err != nil {
+			hc.App.GetLogger().Errorw("Failed to write HTML health report", "err", err)
+			c.AbortWithStatus(http.StatusInternalServerError)
+		}
+		return
+
+	case gin.MIMEPlain:
+		slices.SortFunc(checks, func(a, b presenters.Check) int {
+			return cmp.Compare(a.Name, b.Name)
+		})
+
+		if err := writeTextTo(c.Writer, checks); err != nil {
+			hc.App.GetLogger().Errorw("Failed to write plaintext health report", "err", err)
+			c.AbortWithStatus(http.StatusInternalServerError)
+		}
+		return
+	}
 
 	// return a json description of all the checks
-	jsonAPIResponse(c, checks, "checks")
+	jsonAPIResponseWithStatus(c, checks, "checks", status)
+}
+
+func writeTextTo(w io.Writer, checks []presenters.Check) error {
```

Comment:
why is all this extra new stuff in this change? is it needed to fix the evm healthreport? it seems like a feature that is orthogonal to the health fix. what is the feature getting us relative to what's already implemented?

Reply:
This was the feature that exposed the problem. I can separate the fix, but we don't have good existing tests, so I'd have to write an intermediate test that worked differently.

Reply:
I split out the fix as #11558